### PR TITLE
Add Cat Petting Feature

### PR DESCRIPTION
**Pull request by tylxr59**

### CODE REVIEW

File: oneko.js [hunk 1]
```diff
@@ -196,6 +196,50 @@
     idleAnimationFrame += 1;
   }
 
+  function explodeHearts() {
+    console.log('explodeHearts');
```
Comment (Commenter25) [inline oneko.js:200]: Is this console.log still supposed to be there?

Comment (tylxr59) [reply oneko.js:200]: No, it's not. That's some left over code from my debugging, earlier. I'll fix this as well as well as another section that was missing in one moment

**Comment by tylxr59** [comment]
This version can be viewed on my site - tylxr.com

**Comment by dualfighter1** [comment]
There is a bug where the location of the cat and hearts can desync, I managed to consistently do this by using the back arrow to return to the page from another page, causing the cat to quickly move targets. I  apologise but that is the most accurate I could get the description of it.
image

**Comment by tylxr59** [comment]
@dualfighter1 I made some changes on my site, can you see if that still happens now? I wasn't able to reproduce the bug myself

**Comment by dualfighter1** [comment]
I can still reproducce the issue, my exact steps are:

1. Have the cat moving down
2. Click the visitors link at the bottom of your page
3. Click the back button on my browser (Firefox)
4. Have the cat start moving up again

**Comment by tylxr59** [comment]
Super strange. I tried a couple different ways but couldn't reproduce the bug. I'm on Firefox Dev Edition 126.0b9. I wonder if it's something with the getBoundingClientRect.

If you'd entertain me, could you reproduce it again and provide the output of your Console log in Dev Tools? Should look something like this

image

**Comment by dualfighter1** [comment]
I figured it out, its a bug with scrolling, the hearts dont move while scrolling yet the cat does.

**Comment by tylxr59** [comment]
Ahhh, that makes more sense. I'll try to fix that. tyvm for the feedback

**Comment by dualfighter1** [comment]
No problem.

**Comment by adryd325** [comment]
I'm probably not gonna be merging any new features since its meant to be a simple base for people to build off of, but I'm happy to link your fork in the readme

**Comment by tylxr59** [comment]
For sure! I totally understand. Thatd be awesome if you'd want to link back to my repo :)

**Pull request closed without merging**